src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == arrayList.length) {
+            arrayListGrow();
+        }
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index low than 0 or bigger than array size");
+        }
+        if (size == arrayList.length) {
+            arrayListGrow();
+        }
+        System.arraycopy(arrayList, index, arrayList, index + 1, size - index);
+        arrayList[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index low than 0 or bigger than array size");
+        }
+        return (T) arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index low than 0 or bigger than array size");
+        }
+        arrayList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index low than 0 or bigger than array size");
+        }
+        T removedElement = (T) arrayList[index];
+        System.arraycopy(arrayList, index + 1, arrayList, index, size - index - 1);
+        size--;
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = 0;
+        for (int i = 0; i < size; i++) {
+            if (arrayList[i] == element || arrayList[i] != null && arrayList[i].equals(element)) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("There is not such element");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private T[] arrayListGrow() {
+        T[] newArrayList = (T[]) new Object[(int)((arrayList.length + (arrayList.length * 2)) / 2)];
```

Comment:
```suggestion
        Object[] newArrayList = new Object[(int)(size + size * 1.5)];
        OR
        Object[] newArrayList = new Object[(int)(arrayList.length + arrayList.length * 1.5)];
```
Alos aftrer changing the type of array, no need to cast, and maybe you can change capacity increase to this kind of solution, up to you ^_^

Reply:
if we use 1.5, then we need to add this to private fields as constant, that's why I choose just formula)))

Reply:
Okay, great! :)